### ISSUE TITLE
Portable Windows client?

### ISSUE THREAD

**Issue by ilmaisin** [haiwen/seafile-client#846]
Would it be feasible to release a Seafile portable client that could be installed on an USB or network drive and used on shared workstations that one doesn't have admin rights to?

**Comment by lins05** [comment]
I think you can simply copy the folder "c:\Program Files\Seafile\bin" to a usb drive. To run the client, simply double click the "seafile-applet.exe" file.

**Comment by ilmaisin** [comment]
But where are the settings stored then?

**Comment by ilmaisin** [comment]
https://github.com/naderi/owncloud-client-portable

Naderi has created a portable ownCloud client. Perhaps similar methods could be used for Seafile as well.

**Comment by shoeper** [comment]
In the users home directory. A "real" portable version would require a specially compiled version with changed config paths. In addition using it on Flash Media would reduce their healthyness when syncing large amounts of data.

**Comment by ilmaisin** [comment]
I have now experimented with yaP and Seafile client. A problem I encountered was that Seafile apparently doesn't accept the full USB ID as the path, but requires drive letters. That will be a problem if the drive letter changes from machine to another.

**Comment by shoeper** [comment]
Yes, the client currently is definitely not being designed to be used on a portable device / usb stick. Not sure what `yaP` should be.

**Comment by ilmaisin** [comment]
@shoeper [yaP](http://rolandtoth.hu/yaP/) is a software that is meant to help to make apps "definitely not being designed to be used on a portable device" to run from USB drives. A synchronization client like this however seems to be a rather tricky case. While it probably would be _possible_ to e. g. make a script that changes the paths in Seafile configuration to match the correct ones, editing the app source code itself would likely be more reasonable since Seafile is open source.

**Comment by albcunha** [comment]
This portable version would be very helpful when you need someone send I largefile. You could config a portable seafile client, send it to the costumer and ask him to add the folder where the file is. The webserver/seahub is not a good solution on this case as, if the connection fails, you cant resume upload.

**Comment by condevit** [comment]
ilmaisin mentioned to have solution that only has a problem with the usb drive letter. You can detect the drive letter by the following Batch script....May it help or could you send me your solution so I can try by myself?
Detect usb Driver letter:
set p=%~dp0
echo %p%
set unc=%p::\=:\%
echo %unc%
pause

**Comment by condevit** [comment]
I've made the major works, but w/o changing the source code! Now, I've to do the final testing and packaging...just mail to t.schardt@gmx.de, if you're interested...
